gcsfs/core.py

```diff
@@ -514,13 +692,37 @@ def exists(self, path):
             return False
 
     def info(self, path):
-        path = '/'.join(split_path(path))
-        files = self.ls(path, True)
-        out = [f for f in files if f['name'] == path]
-        if out:
-            return out[0]
-        else:
-            raise FileNotFoundError(path)
+        bucket, key = split_path(path)
+        if not key:
+            # Return a pseudo dir for the bucket root
+            return {
+                'bucket': bucket,
+                'name': "/",
```

Comment:
This may be what a directory entry looks like, but in user-facing methods, the name shoulod be expanded, in this case to 'bucket/'